src/libexpr/get-drvs.cc

```diff
@@ -102,7 +102,21 @@ StorePath DrvInfo::queryOutPath() const
 }
 
 
-DrvInfo::Outputs DrvInfo::queryOutputs(bool onlyOutputsToInstall)
+std::string DrvInfo::queryDefaultOutput()
```

Comment:
I guess the semantics are slightly different? If you implement `defaultOutput` as `drv.outputName`, you're relying on `drv` to already be the default output (and not something like `drv.foo`). I don't know if that's a reasonable assumption in this context, so it seemed clearer to make a separate function.

Reply:
That sounds rather sensible to me. Evaluating `glib.out` is morally equivalent to `glib` in which the default output is `out` (and the order of the outputs shouldn’t really matter besides initially setting `outputName`). So having it show `"defaultOutput": "out"` seems to be the right choice.

(now this is all very confusing in the context of `nix-env` because it will happily ignore all that in favor of `meta.outputsToInstall`. But that’s another matter)

Reply:
OK; I suppose we can rename the field to `outputName` then too to avoid confusion.

Reply:
Yup’ indeed, that’s even better